File: src/main/java/core/procedures/sets/IsSuperset.java
package core.procedures.sets;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

import java.util.Set;

public final class IsSuperset extends AFn {

  public IsSuperset() {
    super(new FnArgsBuilder().min(2).max(2).mandatory(new Class[] {Set.class, Set.class}).build());
  }

  @Override
  public boolean isPure() {
    return true;
  }

  @Override
  public String getName() {
    return "superset?";
  }

  @Override
  public Boolean apply2(Object set1, Object set2) {
    return ((Set)set1).containsAll((Set)set2);
  }
}

File: src/main/java/core/procedures/sets/IsSuperset.kt
package core.procedures.sets

import core.procedures.AFn
import core.procedures.Arity.Exactly

class IsSuperset : AFn<Set<*>?, Boolean>(name = "superset?", isPure = true, arity = Exactly(2),
                                         mandatoryArgsTypes =  arrayOf(Set::class.java, Set::class.java)) {

    override operator fun invoke(arg1: Set<*>?, arg2: Set<*>?) = arg1!!.containsAll((arg2!!))
}
